### ISSUE TITLE
TypeScript: Create new story package with typings

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/web-stories-wp#12090]
These will end up being used by almost all packages and thus cannot really live within `story-editor` because of circular dependency issues.

For example `getStoryMarkup()` in the output package needs it as well (which is a dependency of `story-editor`). See #12089

**Comment by swissspidy** [comment]
That is a good point 👍 

Not sure why I ended up marking this one as being blocked by `animation`  🤔 

I guess I was mostly looking at the `package.json` dependencies and the like and didn't see that the `animation` had so many implicit dependencies on elements, element boxes and resources etc. 

If we only include types in this package, I suppose we could start calling it `@googleforcreators/types` for now and then go from there.
